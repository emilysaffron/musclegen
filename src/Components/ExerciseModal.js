import styled from "@emotion/styled";
import RepCounter from "./RepCounter";
import Confirmation from "./Confirmation";
import { useState } from "react";
const StyledModal = styled.div`
  width: 600px;
  height: 600px;
  background-color: rgba(0, 0, 0, 0.9);
  color: white;
  position: fixed;
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const ExerciseModal = ({
  modal,
  exercise,
  toggleModal,
  AddToWorkoutPlan,
  workoutPlan,
  
}) => {
  const [repNumber, updateRepNumber] = useState(0);
  return modal ? (
    <StyledModal>
      {exercise}
      <RepCounter repNumber={repNumber} updateRepNumber={updateRepNumber} />
      <Confirmation
        choice="cancel"
        repNumber={repNumber}
        exercise={exercise}
        modal={modal}
        toggleModal={toggleModal}
        AddToWorkoutPlan={AddToWorkoutPlan}
        workoutPlan={workoutPlan}
       
      />
    </StyledModal>
  ) : null;
};
export default ExerciseModal;
