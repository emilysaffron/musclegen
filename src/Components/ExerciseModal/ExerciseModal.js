import styled from "@emotion/styled";
import RepCounter from "../RepCounter/RepCounter";
import Confirmation from "../Confirmation/Confirmation";
import { useState } from "react";
const StyledModal = styled.div`
  width: 600px;
  height: 600px;
  background-color: rgba(0, 0, 0, 0.9);
  color: white;
  position: fixed;
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const ExerciseModal = ({
  modal,
  exercise,
  toggleModal,
  AddToWorkoutPlan,
  workoutPlan,
}) => {
  const [repNumber, updateRepNumber] = useState(0);
  return modal ? (
    <StyledModal>
      {exercise}
      <RepCounter repNumber={repNumber} updateRepNumber={updateRepNumber} />
      <Confirmation
        repNumber={repNumber}
        exercise={exercise}
        modal={modal}
        toggleModal={toggleModal}
        AddToWorkoutPlan={AddToWorkoutPlan}
        workoutPlan={workoutPlan}
      />
    </StyledModal>
  ) : null;
};
export default ExerciseModal;
