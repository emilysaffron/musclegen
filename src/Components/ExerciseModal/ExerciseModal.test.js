import renderer from "react-test-renderer";
import ExerciseModal from "./ExerciseModal";

it("ExerciseModal Snapshot", () => {
  const tree = renderer
    .create(
      <ExerciseModal
        modal={true}
        exercise=""
        toggleModal="toggleModal"
        AddToWorkoutPlan="AddToWorkoutPlan"
        workoutPlan="workoutPlan"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
