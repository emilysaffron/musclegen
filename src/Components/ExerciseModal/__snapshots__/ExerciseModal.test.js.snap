// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseModal Snapshot 1`] = `
<div
  className="css-64ccug"
>
  
  <div
    className="css-k46vyb"
  >
    <div
      className="css-ov1ktg"
    >
      <button
        className="css-ikh6w2"
        onClick={[Function]}
      >
        Increase Reps
      </button>
      <button
        className="css-ikh6w2"
        onClick={[Function]}
      >
        Decrease Reps
      </button>
    </div>
    <div
      className="css-nu9ioa"
    />
  </div>
  <div
    className="css-ov1ktg"
  >
    <button
      className="css-ljtnwa"
      onClick={[Function]}
    >
      add to plan
    </button>
    <button
      className="css-ljtnwa"
      onClick={[Function]}
    >
      cancel
    </button>
  </div>
</div>
`;
